lib/ecto/query/builder.ex

```diff
@@ -162,6 +162,24 @@ defmodule Ecto.Query.Builder do
   end
 
   # fragments
+  def escape({:sigil_f, _sigil_meta, [{:<<>>, _meta, pieces}, _modifiers]}, given_type, params_acc, vars, env) do
+    query =
+      pieces
+      |> Enum.map(fn
+        "" <> binary -> binary
```

Comment:
Would this read better or more idiomatically as `binary when is_binary(binary) -> binary`? Or is there a semantic difference in this approach that I am missing?

Reply:
To be honest, I haven't thought much about that, just did the "fastest" way to pattern match on a binary. I don't have a strong opinion about that.